src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,136 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
- * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
- * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
- */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int INITIAL_CAPACITY = 16;
+    private int threshold;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+        capacity = INITIAL_CAPACITY;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (isEnoughSpace(size) == false) {
+            resize(capacity);
+        }
+        if (key == null) {
+            putForNullKey(value);
+        } else {
+            int h = hash(key);
+            int index = indexFor(h, capacity);
+            Node<K, V> nodes = table[index];
+            while (nodes != null) {
+                if (nodes.key != null && nodes.key.equals(key)) {
+                    nodes.value = value;
+                    return;
+                }
+                nodes = nodes.next;
+            }
+            addNode(key, value, index);
+            size++;
+        }
+    }
+
+    private void putForNullKey(V value) {
```

Comment:
do we need this separate method?
let's improve the check: `if (nodes.key != null && nodes.key.equals(key)) ` in the put() method
and then we can remove this method